### PR TITLE
Update embassy to its embedded-* v1.0.0 versions

### LINKED ISSUE #1075: Update embassy to the latest versions
With embedded-* now released, embassy has also been bumped up a version - this means that any embassy features currently do not work.

### PR DESCRIPTION
**Pull request by Sycrosity**
Fixes #1075, by updating Cargo.toml dependencies, replacing embassytime::driver uses with the new embassy-time-driver crate, fixed some miscellanious functions.

I have not yet had time to test all the examples, but all embassy examples except those with pin connections (which havn't been tested) function as expected
Update: embassy_multiprio does not work as of yet. All other embassy examples function as expected.

### Must

- [X] The code compiles without `errors` or `warnings`.
- [x] All examples work.
- [X] `cargo fmt` was run.
- [X] Your changes were added to the `CHANGELOG.md` in the proper section.
- [X] You updated existing examples or added examples (if applicable).
- [x] Added examples are checked in CI

### Nice to have

- [X] You add a description of your work to this PR.
- [X] You added proper docs for your newly added features and code.

### CODE REVIEW

**Comment by jessebraham** [comment]
Thanks for this, would you mind removing all the patches for the `embassy-*` deps from the Cargo manifests as well, please?

**Comment by Sycrosity** [comment]
> Thanks for this, would you mind removing all the patches for the `embassy-*` deps from the Cargo manifests as well, please?

@jessebraham this should now be complete, thanks for pointing this out!

**Comment by bugadani** [comment]
At this time, should the HAL itself depend on embassy-time? I believe the only remaining uses should be examples only?

**Comment by ProfFan** [comment]
> Update: embassy_multiprio does not work as of yet. All other embassy examples function as expected.

Is this still the case? Do you have a description of what does not work?

**Comment by bjoernQ** [comment]
> > Update: embassy_multiprio does not work as of yet. All other embassy examples function as expected.
> 
> Is this still the case? Do you have a description of what does not work?

I think those example have their own problems not really to be addressed in this PR: https://github.com/esp-rs/esp-hal/issues/1080

**Comment by bjoernQ** [comment]
Seems there are still a few patched dependencies:
```
❯ rg 0027a76
esp-hal-smartled\Cargo.toml
33:embassy-executor = { git = "https://github.com/embassy-rs/embassy", rev = "0027a76" }
34:embassy-futures  = { git = "https://github.com/embassy-rs/embassy", rev = "0027a76" }
35:embassy-sync     = { git = "https://github.com/embassy-rs/embassy", rev = "0027a76" }
36:embassy-time     = { git = "https://github.com/embassy-rs/embassy", rev = "0027a76" }

esp-hal-common\Cargo.toml
194:embassy-executor = { git = "https://github.com/embassy-rs/embassy", rev = "0027a76" }
195:embassy-futures  = { git = "https://github.com/embassy-rs/embassy", rev = "0027a76" }
196:embassy-sync     = { git = "https://github.com/embassy-rs/embassy", rev = "0027a76" }
197:embassy-time     = { git = "https://github.com/embassy-rs/embassy", rev = "0027a76" }
```

File: esp-hal-common/Cargo.toml [hunk 1]
```diff
@@ -40,12 +40,13 @@ void                 = { version = "1.0.2", default-features = false }
...
-embassy-executor   = { version = "0.4.0", optional = true }
-embassy-futures    = { version = "0.1.1", optional = true }
-embassy-sync       = { version = "0.5.0", optional = true }
-embassy-time       = { version = "0.2.0", optional = true }
+embedded-hal-async   = { version = "1.0.0", optional = true }
+embedded-io-async    = { version = "0.6.1", optional = true }
+embassy-executor     = { version = "0.5.0", optional = true }
+embassy-futures      = { version = "0.1.1", optional = true }
+embassy-sync         = { version = "0.5.0", optional = true }
+embassy-time         = { version = "0.3.0", optional = true }
```
Comment (MabezDev) [inline esp-hal-common/Cargo.toml:48]:
```suggestion
```

We should only depend on the embassy-time-driver in the hal crates themselves.

File: esp32-hal/Cargo.toml [hunk 2]
```diff
@@ -22,13 +22,13 @@ categories = [
 
 [dependencies]
 esp-hal-common = { version = "0.14.0", features = ["esp32"], path = "../esp-hal-common" }
-embassy-time   = { version = "0.2.0",  optional = true }
+embassy-time   = { version = "0.3.0",  optional = true }
```
Comment (MabezDev) [inline esp32-hal/Cargo.toml:25]: Move this to a dev-dependencies (and apply to all hals)

**Review by MabezDev — CHANGES_REQUESTED** [review]

**Comment by Sycrosity** [comment]
> Seems there are still a few patched dependencies:

@bjoernQ These should be gone now.

**Comment by Sycrosity** [comment]
> > At this point, should the HAL itself depend on embassy-time? I believe the only remaining uses should be examples only?
> 
> I guess yes - good idea! We'll need to remove the `embassy-generic-queue-*` features - the binary crate would need to enable them on their dependency on embassy-time then. Not sure if we should do it in this PR or in a follow up PR

From my testing at least, it is possible to use the `embassy-generic-queue-*` features without the embassy-time dependency - when running: 
```bash
c run --example embassy_hello_world --no-default-features -F embassy,embassy-executor-thread,embassy-time-timg0,embassy-generic-queue-128  -F rt,vectored,xtal-40mhz
```

there are no problems - is there something else I am not aware of preventing this feature?

**Comment by Sycrosity** [comment]
> Thanks!
> 
> `esp-hal-common` still depends on `embassy-time` which is why `embassy-generic-queue-*` just works.
> 
> It would be good to get rid of that dependency in `esp-hal-common` (and those features) - a user can enable the `embassy-generic-queue-*` feature in their own binary crate on their dependency to `embassy-time`

Oh I see - I hadn't realised that. Would that be best done in another issue/PR instead of this one?

**Comment by bjoernQ** [comment]
> > Thanks!
> > `esp-hal-common` still depends on `embassy-time` which is why `embassy-generic-queue-*` just works.
> > It would be good to get rid of that dependency in `esp-hal-common` (and those features) - a user can enable the `embassy-generic-queue-*` feature in their own binary crate on their dependency to `embassy-time`
> 
> Oh I see - I hadn't realised that. Would that be best done in another issue/PR instead of this one? Should I undo the commit removing the dependencies from the esp32-*-hal crates, or leave it here regardless?

Doing that in a separate PR would be completely fine for me. The changes to the `esp32*-hal` crates can be kept I guess

**Review by jessebraham — APPROVED** [review]
We can deal with `embassy-time` in another PR yeah, I will be doing some cleanup before the next release anyway so I can probably just take care of it.

LGTM, thanks for taking care of this!

**Review by MabezDev — APPROVED** [review]
LGTM, thanks!

**Pull request merged**